src/ports/catalog/queries.ts

```diff
@@ -140,22 +100,42 @@ const getItemIdsByTagOrNameQuery = (schemaVersion: string, filters: CatalogQuery
   return query
 }
 
-export const getItemIdsBySearchTextQuery = (schemaVersion: string, filters: CatalogQueryFilters) => {
-  const utilityQuery = getItemIdsByUtilityQuery(schemaVersion, filters)
-  const tagOrNameQuery = getItemIdsByTagOrNameQuery(schemaVersion, filters)
+const getLatestMetadataJoin = (filters: CatalogQueryFilters) => {
+  return filters.network === Network.ETHEREUM
+    ? SQL`
+        LEFT JOIN latest_metadata ON latest_metadata.item_id = items.metadata ` // TODO: This will be fix during next indexation, is a workaround for the current one
+    : SQL`
+        LEFT JOIN latest_metadata ON latest_metadata.item_id = items.id `
+}
+
+const getLatestMetadataCTE = () => {
+  return SQL`latest_metadata AS (
+        SELECT DISTINCT ON (wearable_id) wearable_id as item_id, id AS latest_metadata_id, item_type, wearable_id, emote_id
+        FROM metadata
+        ORDER BY wearable_id DESC
+      )
+    `
+}
 
-  return SQL`SELECT DISTINCT ON (id) id,
-      word_similarity,
-      match_type,
-      word
-      FROM ((`
-    .append(utilityQuery)
-    .append(SQL`) UNION (`)
-    .append(tagOrNameQuery).append(SQL`)) AS items_found
-      ORDER BY id DESC, word_similarity DESC
-    `)
+const getSearchCTEs = (filters: CatalogQueryFilters) => {
+  return SQL`WITH `.append(getLatestMetadataCTE()).append(
+    SQL`, builder_server_items AS (
+      SELECT
+      item_id,
+      tag
+    FROM
+      `.append(BUILDER_SERVER_TABLE_SCHEMA).append(SQL`.mv_builder_server_items
+    WHERE
+      LOWER(tag) = LOWER(${filters.search})
+    )
+  `)
+  )
 }
 
+const WEARABLE_ITEM_TYPES = [FragmentItemType.WEARABLE_V1, FragmentItemType.WEARABLE_V2, FragmentItemType.SMART_WEARABLE_V1]
+
+const MAX_ORDER_TIMESTAMP = 253378408747000 // some orders have a timestmap that can't be cast by Postgres, this is the max possible value
```

Comment:
Could this be fixable by using numeric or will it impact heavily on the queries performance?